Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoft365Defender.yaml

```diff
@@ -47,33 +49,37 @@ ParserQuery: |
       EventResult = iff(ActionType=='ConnectionFailed','Failure','Success'),
       EventSeverity = "Informational",
       DvcIdType = 'MDEid'
-    | project-rename
-      NetworkProtocol = Protocol,
-      DvcFQDN = DeviceName
+    | extend
+      RemoteUrl = extract (@"(?:https?://)?(.*)", 1, RemoteUrl)
     | extend
       User = iff (InitiatingProcessAccountDomain == '', InitiatingProcessAccountName, strcat(InitiatingProcessAccountDomain, '\\', InitiatingProcessAccountName)),
       UsernameType = iff(InitiatingProcessAccountDomain == '','Simple', 'Windows'),
-      SplitHostname = split(DvcFQDN,"."),
-      SplitUrl = split(RemoteUrl,".")
+      SplitHostname = split(DeviceName,"."),
+      SplitUrl = split(RemoteUrl,"."),
+      NetworkProtocol = case (
+        Protocol startswith "Tcp", "TCP",
+        Protocol == "Unknown", "",
+        toupper(Protocol)
+      )
     | extend  
       DvcHostname = tostring(SplitHostname[0]),
       DvcDomain = strcat_array(array_slice(SplitHostname, 1, -1), '.'),
+      DvcFQDN = iif (DeviceName contains ".", DeviceName, ""),
       UrlHostname = SplitUrl[0],
-      UrlDomain = strcat_array(array_slice(SplitUrl, 1, -1), '.'),
-      SrcDomainType = "FQDN",
-      DvcDomainType = "FQDN",
-      DstDomainType = "FQDN",
+      UrlDomain = tostring(strcat_array(array_slice(SplitUrl, 1, -1), '.')),
+      UrlFQDN = iif(RemoteUrl contains ".", RemoteUrl, "")
+    | extend
+      DvcDomainType = iif(DvcFQDN != "", "FQDN", ""),
+      UrlDomainType = iff(UrlFQDN != "", "FQDN", ""),
```

Comment:
I am not sure the method for parsing Host works for Url. for example contoso.com -  Url domain would be "com"

Reply:
first com is a domain.... there is no easy way around this. Essentially you have to do a lookup for known TLDs. I will leave it as is.

Reply:
ok. maybe we need to push this as a kusto request, more sophisticated and accurate extraction of the domain?